Description: Make db-manage script use same config file as barbican
 Instead of having to specify the db-url every time we need to do a migration,
 this CR enables the db-manage script to use the same config file as barbican.
Author: Juan Antonio Osorio Robles <juan.osorio.robles@ericsson.com>
Origin: upstream, https://review.openstack.org/#/c/194665/
Last-Update: 2015-06-23

--- barbican-2015.1.0.orig/bin/barbican-db-manage.py
+++ barbican-2015.1.0/bin/barbican-db-manage.py
@@ -17,7 +17,8 @@ class DatabaseManager:
     using Alembic commands.
     """
 
-    def __init__(self):
+    def __init__(self, conf):
+        self.conf = conf
         self.parser = self.get_main_parser()
         self.subparsers = self.parser.add_subparsers(
             title='subcommands',
@@ -31,7 +32,7 @@ class DatabaseManager:
     def get_main_parser(self):
         """Create top-level parser and arguments."""
         parser = argparse.ArgumentParser(description='Barbican DB manager.')
-        parser.add_argument('--dburl', '-d', default=None,
+        parser.add_argument('--dburl', '-d', default=self.conf.sql_connection,
                             help='URL to the database.')
 
         return parser
@@ -126,12 +127,12 @@ def main():
     # Import and configure logging.
     CONF = cfg.CONF
     log.register_options(CONF)
-    log.setup(CONF, 'barbican-db-manage')
+    log.setup(CONF, 'barbican')
     LOG = log.getLogger(__name__)
     LOG.debug("Performing database schema migration...")
 
     try:
-        dm = DatabaseManager()
+        dm = DatabaseManager(CONF)
         dm.execute()
     except Exception as ex:
         if _exception_is_successfull_exit(ex):
